asserts/asserts.go

```diff
@@ -887,22 +898,41 @@ func (enc *Encoder) append(encoded []byte) error {
 		return err
 	}
 
-	if encoded[sz-1] != '\n' {
-		_, err = enc.wr.Write(nl)
-		if err != nil {
-			return err
-		}
+	return enc.writeSep(encoded[sz-1])
+}
+
+// WriteContentSignature writes the content and signature of an assertion into the stream with all the required separators.
+func (enc *Encoder) WriteContentSignature(content, signature []byte) error {
+	sz := len(signature)
+	if sz == 0 {
+		return fmt.Errorf("internal error: signature cannot be empty")
 	}
```

Comment:
To be clear, no `content` will not break the internals of writing the stream. However, all assertions will have some content so it's definitely an error. However, it's really the developer's fault so adding the extra check is not absolutely necessary, just nice.

So, will +1 the PR and let you decide.